src/python/grpcio_observability/grpc_observability/_cyobservability.pyx

```diff
@@ -0,0 +1,249 @@
+# Copyright 2023 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cimport cpython
+from cython.operator cimport dereference
+
+import logging
+from threading import Thread
+from typing import List, Mapping, Tuple
+
+import grpc_observability
+
+
+cdef const char* CLIENT_CALL_TRACER = "client_call_tracer"
+cdef const char* SERVER_CALL_TRACER_FACTORY = "server_call_tracer_factory"
+cdef bint GLOBAL_SHUTDOWN_EXPORT_THREAD = False
+cdef object global_export_thread
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MetricsName:
+  CLIENT_API_LATENCY = kRpcClientApiLatencyMeasureName
+  CLIENT_SNET_MESSSAGES_PER_RPC = kRpcClientSentMessagesPerRpcMeasureName
+  CLIENT_SEND_BYTES_PER_RPC = kRpcClientSentBytesPerRpcMeasureName
+  CLIENT_RECEIVED_MESSAGES_PER_RPC = kRpcClientReceivedMessagesPerRpcMeasureName
+  CLIENT_RECEIVED_BYTES_PER_RPC = kRpcClientReceivedBytesPerRpcMeasureName
+  CLIENT_ROUNDTRIP_LATENCY = kRpcClientRoundtripLatencyMeasureName
+  CLIENT_SERVER_LATENCY = kRpcClientServerLatencyMeasureName
+  CLIENT_STARTED_RPCS = kRpcClientStartedRpcsMeasureName
+  CLIENT_RETRIES_PER_CALL = kRpcClientRetriesPerCallMeasureName
+  CLIENT_TRANSPARENT_RETRIES_PER_CALL = kRpcClientTransparentRetriesPerCallMeasureName
+  CLIENT_RETRY_DELAY_PER_CALL = kRpcClientRetryDelayPerCallMeasureName
+  CLIENT_TRANSPORT_LATENCY = kRpcClientTransportLatencyMeasureName
+  SERVER_SENT_MESSAGES_PER_RPC = kRpcServerSentMessagesPerRpcMeasureName
+  SERVER_SENT_BYTES_PER_RPC = kRpcServerSentBytesPerRpcMeasureName
+  SERVER_RECEIVED_MESSAGES_PER_RPC = kRpcServerReceivedMessagesPerRpcMeasureName
+  SERVER_RECEIVED_BYTES_PER_RPC = kRpcServerReceivedBytesPerRpcMeasureName
+  SERVER_SERVER_LATENCY = kRpcServerServerLatencyMeasureName
+  SERVER_STARTED_RPCS = kRpcServerStartedRpcsMeasureName
+
+# Delay map creation due to circular dependencies
+_CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING = {}
+
+def cyobservability_init(object exporter) -> None:
+  global _CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING
+  _CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING = {x.value: x for x in grpc_observability.MetricsName}
+  NativeObservabilityInit()
+  _start_exporting_thread(exporter)
+
+
+def _start_exporting_thread(object exporter) -> None:
+  global global_export_thread
+  global GLOBAL_SHUTDOWN_EXPORT_THREAD
+  GLOBAL_SHUTDOWN_EXPORT_THREAD = False
```

Comment:
Is it necessary to set this here? This is already the default value.

Reply:
I think it's better to keep this line, for the use case that user tries to enable observability again after exit (also in testing where we import `grpc_observability` once and need reinitiate it multiple times across different test cases).